info.py

```diff
@@ -28,45 +30,54 @@ def print(self, *args, **kwargs):
         sys.stdout.write(chunk)
         self.chunks.extend(chunk)
 
-    def heading(self, name):
+    def heading(self, name, level):
         self.print()
         self.print(name)
-        self.print("=" * len(name))
+        self.print(self.heading_marks[level] * len(name))
         self.print()
 
     def value(self):
         return "".join(chunk for chunk in self.chunks)
 
+    def print_mapping(self, mapping):
+        maximum_key_length = max(len(repr(key)) for key in mapping.keys())
+
+        for name, value in sorted(mapping.items()):
+            line = "{: <{}} : {}".format(
+                repr(name),
+                maximum_key_length,
+                repr(value),
+            )
+            self.print(line)
+
+
 output = Output()
 
-output.heading("Python Details")
+environment = dict(os.environ)
+
+context_prefix = "_PYTHON_INFO_ACTION_CONTEXT_"
```

Comment:
Maybe rename to uppercase and move it at the start of the file.

This looks like a configuration value for an external decency.

```suggestion
CONTEXT_PREFIX = "_PYTHON_INFO_ACTION_CONTEXT_"
```

Reply:
I've read and written too many C macros and thus have a visceral reaction to `YELLING_HACKY_GARBAGE_AUAAUUGHHGHGH`...  but yes, that isn't what it is in Python and it is common.